ui/src/core_components/embed/CorePDF.vue

```diff
@@ -33,26 +60,251 @@ export default {
+		gotoHighlight(currentMatch.value);
+	}
+};
+
+const scroll = (event) => {
+	const c = event.target.getBoundingClientRect();
+	const r = [...event.target.children]
+		.filter((child) => child.className.includes("page"))
+		.find((child) => {
+			const e = child.getBoundingClientRect();
+			const relativeY = e.y - c.y;
+			return (
+				relativeY < c.height / 2 && relativeY + e.height > c.height / 2
+			);
+		});
+	if (r) {
+		page.value = Number(r.getAttribute("page"));
+	}
+};
+
+const calcScrollPosition = (targetEl: HTMLElement, parentEl: HTMLElement) => {
+	let offsetTop = 0;
+	let el = targetEl;
+	while (el && el !== parentEl) {
+		offsetTop += el.offsetTop;
+		el = el.offsetParent as HTMLElement;
+	}
+	return offsetTop;
+};
+
+const gotoHighlight = (matchIdx: number) => {
+	if (matchIdx < 1 || matchIdx > matches.value.length) {
+		return;
+	}
+	const match = matches.value[matchIdx - 1];
+	console.log(match);
+	const matchEls = rootEl.value.querySelectorAll(
+		`div[page='${match.page}'] span.highlight`,
+	);
+	const matchEl = matchEls[match.index];
+	// scrollIntoView is breaking pdf viewer
+	viewerEl.value.scrollTop = calcScrollPosition(matchEl, viewerEl.value);
+};
+
+const gotoPage = (page: number) => {
+	if (page < 1 || page > pages.value) {
+		return;
+	}
+	const pageEl = rootEl.value.querySelector("div[page='" + page + "']");
+	viewerEl.value.scrollTop = calcScrollPosition(pageEl, viewerEl.value);
+};
+
+const onLoaded = () => {
+	if (fields.page.value) {
+		gotoPage(fields.page.value);
+	}
+	if (fields.selectedMatch.value) {
+		currentMatch.value = fields.selectedMatch.value;
+	}
+};
+
+const incrementMatchIdx = () => {
+	if (currentMatch.value < matches.value.length) {
+		currentMatch.value = currentMatch.value + 1;
+	} else {
+		currentMatch.value = 1;
+	}
+};
+
+const decrementMatchIdx = () => {
+	if (currentMatch.value > 1) {
+		currentMatch.value = currentMatch.value - 1;
+	} else {
+		currentMatch.value = matches.value.length;
+	}
+};
+
+const incrementScale = () => {
+	scale.value = scale.value < 2 ? scale.value + 0.1 : scale.value;
+};
+
+const decrementScale = () => {
+	scale.value = scale.value > 0.25 ? scale.value - 0.1 : scale.value;
+};
+
+watch(fields.source, () => {
+	matches.value = [];
+});
+
+watch(fields.highlights, () => {
+	matches.value = [];
+});
+
+watch(currentMatch, () => {
+	gotoHighlight(currentMatch.value);
+});
+
+watch(fields.selectedMatch, () => {
+	currentMatch.value = fields.selectedMatch.value;
+});
+
+watch(fields.page, () => {
+	gotoPage(fields.page.value);
+});
 </script>
 
 <style scoped>
 @import "../../renderer/sharedStyles.css";
+
 .CorePDF {
 	position: relative;
+	display: flex;
+	flex-direction: column;
 	width: 100%;
 	height: 80vh;
 }
 
+.CorePDF .controls {
+	flex: 0 0 40px;
+	display: flex;
+	flex-direction: row;
+	justify-content: space-between;
+	align-items: center;
+	background-color: var(--backgroundColor);
+	border-bottom: 1px solid var(--separatorColor);
```

Comment:
Please also add

`color: var(--primaryTextColor);`

For the color to be reflected (should happen in most cases but wouldn't rely on it)

Reply:
Also, play around with the colours and see how it goes, should render a decent image regardless of using default colours or not.

Reply:
👍